Snowflake.Data/Core/FileTransfer/StorageClient/SFGCSClient.cs

```diff
@@ -122,68 +140,33 @@ public FileHeader GetFileHeader(SFFileMetadata fileMetadata)
-                        var contentLength = response.Headers.GetValues("content-length");
+                    var digest = response.Headers.GetValues(GCS_METADATA_SFC_DIGEST);
+                    var contentLength = response.Headers.GetValues("content-length");
 
-                        fileMetadata.resultStatus = ResultStatus.UPLOADED.ToString();
+                    fileMetadata.resultStatus = ResultStatus.UPLOADED.ToString();
 
-                        return new FileHeader
-                        {
-                            digest = digest[0],
-                            contentLength = Convert.ToInt64(contentLength[0])
-                        };
-                    }
-                }
-                catch (WebException ex)
-                {
-                    HttpWebResponse response = (HttpWebResponse)ex.Response;
-                    if (response.StatusCode == HttpStatusCode.Unauthorized ||
-                        response.StatusCode == HttpStatusCode.Forbidden ||
-                        response.StatusCode == HttpStatusCode.NotFound)
+                    return new FileHeader
                     {
-                        fileMetadata.resultStatus = ResultStatus.NOT_FOUND_FILE.ToString();
-                        return new FileHeader();
-                    }
+                        digest = digest[0],
+                        contentLength = Convert.ToInt64(contentLength[0])
+                    };
                 }
             }
-            else
+            catch (WebException ex)
             {
-                // Generate the file URL based on GCS location
-                string url = generateFileURL(fileMetadata.stageInfo.location, fileMetadata.RemoteFileName());
-                try
-                {
-                    // Issue a HEAD request
-                    HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
-                    request.Method = "HEAD";
-                    request.Headers.Add("Authorization", $"Bearer {AccessToken}");
-
-                    using (HttpWebResponse response = (HttpWebResponse)request.GetResponse())
-                    {
-                        var digest = response.Headers.GetValues(GCS_METADATA_SFC_DIGEST);
-                        var contentLength = response.Headers.GetValues("content-length");
-
-                        fileMetadata.resultStatus = ResultStatus.UPLOADED.ToString();
-
-                        return new FileHeader
-                        {
-                            digest = digest[0],
-                            contentLength = Convert.ToInt64(contentLength[0])
-                        };
-                    }
-                }
-                catch (WebException ex)
-                {
-                    fileMetadata = HandleFileHeaderErr(ex, fileMetadata);
-                }
+                // GCS presigned urls have a different error handling for file headers
```

Comment:
let's log the error from ex

Reply:
Since it calls either `HandleFileHeaderErrForPresignedUrls` or `HandleFileHeaderErrForGeneratedUrls`, I added the logging in those functions instead so it gets logged for async errors as well